students/BrianB/lesson04/trigrams.py

```diff
@@ -8,6 +9,7 @@ def read_file(filename):
     with open('sherlock_small.txt') as text:
         for line in text:
             words.extend(line.replace("--", " ").split())
+            string.punctuation  # '!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~'
```

Comment:
What I was suggesting with string.punctuation was to check if a character exists in string.punctuation. If so, you know it's punctuation and you can replace it with "". So if you have a word, you know punctuation is usually going to be at the beginning or end, so check that character if it exists in string.punctuation.


Ex:  assume you have a word
if word[0] in string.punctuation:
word = word[1:]
if word[-1] in string.punctuation:
word = word[:-1]